src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -435,26 +436,55 @@ impl ScalarFunc {
         ctx: &mut EvalContext,
         row: &'a [Datum],
     ) -> Result<Option<Cow<'a, MyDuration>>> {
-        let d0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
-        let d1: Cow<'a, MyDuration> = try_opt!(self.children[1].eval_duration(ctx, row));
-        let diff = match d0.to_nanos().checked_sub(d1.to_nanos()) {
-            Some(result) => result,
-            None => return Err(Error::overflow("DURATION", &format!("({} - {})", &d0, &d1))),
+        let arg0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
+        let arg1: Cow<'a, MyDuration> = try_opt!(self.children[1].eval_duration(ctx, row));
+        let overflow = Error::overflow("DURATION", &format!("({} - {})", &arg0, &arg1));
```

Comment:
Is it related to current PR?

Reply:
For code consistency, I think it's better.